docs/API.md

```diff
@@ -884,6 +925,29 @@ in Rendezvous connection; the result might be that simply one side will override
 the value from the other side and it's the matter of luck which one would win
 ---
 
+| OptName           | Since | Binding | Type            | Units | Default  | Range  |
+| ----------------- | ----- | ------- | --------------- | ----- | -------- | ------ |
+| `SRTO_STRICTENC`  | 1.3.2 | pre     | `int (bool)`    |       | true     | false  |
+
+- When this option is set (default) then only such connections are allowed that
```

Comment:
- This option defines how strictly encryption is applied at both ends of a connection.

- When this option is set to TRUE, connections are allowed only when:
   -- neither peer has enabled encryption.
   -- both peers have enabled encryption with the same passphrase.

- If one of these conditions is not met, the connection will be rejected.

- When this option is set to FALSE, then the following types of connections
will be allowed (with appropriate limitations):
   -- both peers have enabled encryption with the different passphrases; the connection is allowed, but data transmission will be impossible in either direction.
   -- only one peer has enabled encryption; the connection is allowed, and the peer with encryption enabled will accept unencrypted transmissions.

- Setting `SRTO_STRICTENC` to FALSE can be useful in situations where it is important to know whether a connection is possible. The inability to decrypt an incoming transmission can be reported as a different kind of problem.

Reply:
It is important to explain that STRICTENC shortcut doesn't mean "strict encryption", but "strictly encrypted" - people were often mistaking this term and the first wrong understanding was causing confusion among customers.

What is meant here is that there are actually 4 available cases of encryption combinations, out of which 2 are meant as "strictly encrypted", and the other 2 are "non-strictly-encrypted".

This option changes only one thing: when it's set to true, then "non-strictly-encrypted" combination causes the connection to be rejected. If set to false, the connection is accepted, although the transmission abilities are appropriately impaired.

Reply:
Ok, I redacted it into this:

- **[SET]** - This option defines whether the connection is allowed for only
"strictly encrypted" case.

When this option is set to TRUE (default), connections are allowed only when:
   - neither party has enabled encryption
   - both parties have enabled encryption with the same passphrase

If one of these conditions is not met, the connection will be rejected.

When this option is set to FALSE, then connection for the remaining combinations
("non-stricly-encrypted") is allowed, however the transmission will be
appropriately impaired:

   - both parties have enabled encryption with different passphrase
      - transmission not possible in either direction
   - only one party has enabled encryption
      - unencrypted transmission possible only from unencrypted party to encrypted one

Setting `SRTO_STRICTENC` to FALSE can be useful only for very specific appliances,
where you want to distinguish the problem of making a connection (so you want it to
succeed, even if the transmission would be impaired) and the problem with
encryption, as well as in this particular case allowing for unencrypted stream
transmission isn't treated as insecure.

Reply:
This could also enumerate all combinations first, for example:

This option controls which combinations of encryption setup for both parties are allowed, that is:

1. Both parties unencrypted
2. Both parties encrypted and have the same passphrase
3. Both parties encrypted and have different passphrases
4. One party encrypted, the other unencrypted

When this option is set to TRUE (default), it allows a connection only for "strictly encrypted" cases, that is, 1 and 2. In other cases the connection will be rejected.

When this option is set to FALSE, then the connection in cases 3 and 4 is allowed, but:

3. Transmission will not be possible in either direction
4. Unencrypted transmission will be possible from the unencrypted party to the encrypted party.

Reply:
The distinction between the phrases "strict encryption" and "strictly encrypted" is not clear. I think the text already explains that distinction by describing the options and their consequences. I've made some slight edits and fixed the formatting, but it otherwise remains the same as my previous edit:

```
The `SRTO_STRICTENC`option defines how strictly encryption is applied and/or interpreted at both ends of a connection.

When this option is set to TRUE connections are allowed only when:
<ul>
<li>neither peer has enabled encryption.</li>
<li>both peers have enabled encryption with the same passphrase.</li>
</ul>
If one of these conditions is not met, the connection will be rejected.

When this option is set to FALSE the following types of connections
will be allowed (with appropriate limitations):

<ul>
<li>both peers have enabled encryption with the different passphrases; the connection is allowed, but data transmission will be impossible in either direction.</li>
<li>only one peer has enabled encryption; the connection is allowed, and the peer with encryption enabled will accept unencrypted transmissions.</li>
</ul>

Setting the `SRTO_STRICTENC`option to FALSE can be useful in situations where it is important to know whether a connection is possible. The inability to decrypt an incoming transmission can be reported as a different kind of problem. Note that the option must be set to FALSE on BOTH peers (otherwise, the default value TRUE is applied on both).
```

Reply:
This was quite a problem for me to speak in a few easily understood words, what this option really controls, and moreover, in a way that allows a user to easily remember it. This is the current form:

- **[SET]** - This option, when set to TRUE, allows connection only if the
encryption setup of the connection parties is a "strictly encrypted" case,
that is:
   - neither party has enabled encryption
   - both parties have enabled encryption with the same passphrase

What is important to highlight:
   - "strictly encrypted", the meaningful statement for this option, and whether this is the case, it depends on what settings are set up on both parties together
   - what this option controls is WHETHER it allows for connection in non-strictly-encrypted cases